monitoring/workqueue/metrics.go

```diff
@@ -0,0 +1,123 @@
+package workqueue
+
+import (
+	"k8s.io/client-go/util/workqueue"
+
+	"github.com/longhorn/longhorn-manager/monitoring/registry"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+// Package prometheus sets the workqueue DefaultMetricsFactory to produce
+// prometheus metrics. To use this package, you just have to import it.
+
+// Metrics subsystem and keys used by the workqueue.
+const (
+	LonghornName               = "longhorn"
+	WorkQueueSubsystem         = "workqueue"
+	DepthKey                   = "depth"
+	AddsKey                    = "adds_total"
+	QueueLatencyKey            = "queue_duration_seconds"
+	WorkDurationKey            = "work_duration_seconds"
+	UnfinishedWorkKey          = "unfinished_work_seconds"
+	LongestRunningProcessorKey = "longest_running_processor_seconds"
+	RetriesKey                 = "retries_total"
+)
+
+var (
+	depth = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Namespace: LonghornName,
+		Subsystem: WorkQueueSubsystem,
+		Name:      DepthKey,
+		Help:      "Current depth of workqueue",
+	}, []string{"name"})
+
+	adds = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Namespace: LonghornName,
+		Subsystem: WorkQueueSubsystem,
+		Name:      AddsKey,
+		Help:      "Total number of adds handled by workqueue",
+	}, []string{"name"})
+
+	latency = prometheus.NewHistogramVec(prometheus.HistogramOpts{
+		Namespace: LonghornName,
+		Subsystem: WorkQueueSubsystem,
+		Name:      QueueLatencyKey,
+		Help:      "How long in seconds an item stays in workqueue before being requested.",
```

Comment:
Do you mean `being reconciled` or `being requested`?

Reply:
I meant `being requested`. I copied those definitions from kubernetes package :D

Reply:
IMO, this latency means the interval from the item being enqueued to the item being poped up (for reconciliation), right?

Reply:
> IMO, this latency means the interval from the item being enqueued to the item being poped up 
Yes, I think so too. From the queue package's perspective, the item is poped up. The queue doesn't know what the consumer is doing with it (reconciliation/process/etc...). So I think the queue just knows that the item is requested.